modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx

```diff
@@ -41,6 +42,173 @@ encounter any issues or use cases that we've missed.
+    />
+  }
+/>;
+
+// v8
+import {DeprecatedDrawer, DeprecatedDrawerHeader} from '@workday/canvas-kit-labs-react/drawer';
+
+<DeprecatedDrawer
+  header={
+    <DeprecatedDrawerHeader
+      closeIconAriaLabel="Close"
+      onClose={() => console.log('onClose callback')}
+      title="Deprecated Drawer Header"
+    />
+  }
+/>;
+```
+
+#### Menu (preview)
+
+We are [soft deprecating](#soft-deprecation) `Menu`. It has been renamed to `DeprecatedMenu`.
+Similarly, `MenuProps` has been renamed to `DeprecatedMenuProps`, `MenuState` to
+`DeprecatedMenuState`, `MenuItem` to `DeprecatedMenuItem` and `MenuItemProps` to
+`DeprecatedMenuItemProps`. You may continue to use this component, but note that we plan to
+[hard-deprecate](#hard-deprecation) it in Canvas Kit v9.
+
+🤖 The codemod will rename `Menu`, `MenuProps`, `MenuState`, `MenuItem` and `MenuItemProps` to their
+deprecated names:
+
+```tsx
+// v7
+import {Menu, MenuItem, MenuProps, MenuItemProps} from '@workday/canvas-kit-preview-react/menu';
+
+interface AnotherMenuProps extends MenuProps {
+  specialProp?: boolean;
+}
+type CustomMenuItemProps = MenuItemProps;
+
+const CustomMenu = () => {
+  return (
+    <Menu>
+      <MenuItem>First Item</MenuItem>
+    </Menu>
+  );
+};
+
+// v8
+import {DeprecatedMenu, DeprecatedMenuItem} from '@workday/canvas-kit-preview-react/menu';
+
+interface AnotherMenuProps extends DeprecatedMenuProps {
+  specialProp?: boolean;
+}
+
+type CustomMenuItemProps = DeprecatedMenuItemProps;
+
+const CustomMenu = () => {
+  return (
+    <DeprecatedMenu>
+      <DeprecatedMenuItem>First Item</DeprecatedMenuItem>
+    </DeprecatedMenu>
+  );
+};
+```
+
+#### Layout and Column
+
+We are [soft deprecating](#soft-deprecation) `Layout` and `Column`. It has been renamed to
+`DeprecatedLayout` and `DeprecatedColumn`. Similarly, `LayoutProps` has been renamed to
+`DeprecatedLayoutProps`, `Column` to `DeprecatedColumn` and `ColumnProps` to
+`DeprecatedColumnProps`. You may continue to use this component, but note that we plan to
+[hard-deprecate](#hard-deprecation) it in Canvas Kit v9.
+
+🤖 The codemod will rename `Layout`, `LayoutProps`, `Column` and `ColumnProps` to their deprecated
+names:
+
+```tsx
+// v7
+import {Layout, Column} from '@workday/canvas-kit-react/drawer';
+
+<Layout gutter={0}>
+  <Layout.Column spacing={0}>
+    <Card />
+  </Layout.Column>
+</Layout>;
+
+// v8
+import {DeprecatedLayout, DeprecatedColumn} from '@workday/canvas-kit-react/drawer';
+
+<DeprecatedLayout gutter={0}>
+  <DeprecatedLayout.Column spacing={0}>
+    <Card />
+  </DeprecatedLayout.Column>
+</DeprecatedLayout>;
+```
+
+#### SegmentedControl
+
+We are [soft deprecating](#soft-deprecation) `SegmentedControl` and it has been renamed to
+`DeprecatedSegmentedControl`. Similarly, `SegmentedProps` has been renamed to
+`DeprecatedSegmentedProps`. You may continue to use this component, but note that we plan to
+[hard-deprecate](#hard-deprecation) it in Canvas Kit v9.
+
```

Comment:
Same here, encourage users to use the new compound component one that you're building :)